common/src/main/java/com/microsoft/hydralab/performance/inspectors/AndroidBatteryInfoInspector.java

```diff
@@ -2,15 +2,47 @@
 // Licensed under the MIT License.
 package com.microsoft.hydralab.performance.inspectors;
 
+import com.microsoft.hydralab.common.util.ShellUtils;
+import com.microsoft.hydralab.common.util.TimeUtils;
 import com.microsoft.hydralab.performance.PerformanceInspection;
 import com.microsoft.hydralab.performance.PerformanceInspectionResult;
 import com.microsoft.hydralab.performance.PerformanceInspector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.Assert;
+
+import java.io.File;
 
 public class AndroidBatteryInfoInspector implements PerformanceInspector {
+    private static final String RAW_RESULT_FILE_NAME_FORMAT = "%s_%s_%s.txt";
+    protected Logger classLogger = LoggerFactory.getLogger(getClass());
 
     @Override
     public PerformanceInspectionResult inspect(PerformanceInspection performanceInspection) {
-        return null;
+        if (performanceInspection.isReset) {
+            initialize(performanceInspection);
+        }
+
+        File rawResultFolder = new File(performanceInspection.resultFolder, performanceInspection.appId);
+        Assert.isTrue(rawResultFolder.exists() || rawResultFolder.mkdir(), "rawResultFolder.mkdirs() failed in" + rawResultFolder.getAbsolutePath());
+        File rawResultFile = new File(rawResultFolder,
+                String.format(RAW_RESULT_FILE_NAME_FORMAT, getClass().getSimpleName(), performanceInspection.appId, TimeUtils.getTimestampForFilename()));
+
+
+        ShellUtils.execLocalCommandWithResult(String.format("adb -s %s shell dumpsys batterystats %s | out-file %s -encoding utf8",
+                performanceInspection.deviceIdentifier, performanceInspection.appId, rawResultFile.getAbsolutePath()), classLogger);
+        return new PerformanceInspectionResult(rawResultFile, performanceInspection);
+    }
+
+    private void initialize(PerformanceInspection performanceInspection) {
+        String device = performanceInspection.deviceIdentifier;
+        ShellUtils.execLocalCommand(String.format("adb -s %s shell dumpsys battery unplug", device), classLogger);
+        ShellUtils.execLocalCommand(String.format("adb -s %s shell dumpsys batterystats --reset", device), classLogger);
+    }
+
+    // TODO: Do we need to add the method to the Interface, or add  finalize flag to the PerformanceInspection?
+    public void finalize(PerformanceInspection performanceInspection) {
```

Comment:
Do we need to add the method to the Interface, or add finalize flag to the PerformanceInspection?  As we have to reset the battery status at test run end. cc @dinghong2233 @zhou9584

Reply:
I this depends on whether more parsers use this function currently. If so, I agree to put it into the interface.
The same is for whether to add the finalize flag.

Reply:
Thanks. And if no more inspectors/parsers use this function. Maybe we could tear down the performance related env in `TestRunner` to avoid unexpected exceptions. And it may not be the best practice for inspector to have lifecycle.